### PR TITLE
Chore: Enable auto-merge greenkeeper

### LINKED ISSUE #938: Automate dependency update process
We should have a script that checks for dependency updates and is periodically execute by a cron job. If tests pass the script should automatically update the dependencies (maybe only `devDependencies`?) and commit the changes, otherwise let the automation process open an issue so we can review and fix the problems.

---

See also: #719

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~Added/Updated related documentation.~~
- ~~Added/Updated related tests.~~

## Short description of the change(s)

The script `greenkeeper-merge.js` is executed as a job when all the
tests passed. `greenkeeper` does a commit per each package in the
monorepo that shares the same dependency. This scripts:

* Squashes all of those commits (takes all the commits from the end
  with the same message until one is different).
* Rebases the changes into master and pushes.

**NOTE**: The `install` step of `travis.yml` performs a `yarn`. If the
version is different, the generated `yarn.lock` file will be too. This
script discards any changes in the `yarn.lock` that originate in that
step (`greenkeeper` takes care of updating it).

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Fix #938

### CODE REVIEW

**Comment by molant** [comment]
I've tested this in my fork and seems to be working without any issues after a few trials.
Because this is only executed on `greenkeeper` branches, I think it can be merged before we enable it (although maybe first we want to update the dependencies).

File: .travis.yml [hunk 1]
```diff
@@ -110,6 +113,21 @@ jobs:
...
 
+    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    # Stage used for automatically merging Greenkeeper's commits
+    # that pass the tests.
+    #
+    # [!] Runs: Only for branches created by Greenkeeper.
+
+    - stage: greenkeeper-merge
+      if: type = push and branch =~ ^greenkeeper\/.*-.*$
```
Comment (alrra) [inline .travis.yml]: For consistency, use:
```yml
if: |
      branch =~ ^greenkeeper\/.*-.*$ and \
      type = push
 ```

File: .travis/greenkeeper-merge.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    return null;
+};
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+const main = async () => {
+    const GIT_COMMIT_SUBJECT = await getLastCommitSubject();
+    const COMMITS_TO_SQUASH = await getLastCommitsWithSameMessage();
+    const NEW_COMMIT_MESSAGE = generateNewCommitMessage(GIT_COMMIT_SUBJECT);
+
```
Comment (alrra) [inline .travis/greenkeeper-merge.js]: Remove extra empty line.

File: .travis/greenkeeper-merge.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                `git config --global user.name "${GIT_USER_NAME}"
+                 git config --global user.email "${GIT_USER_EMAIL}"`);
+    // `install` step in travis does a yarn. If versions are different, yarn.lock is different so discard changes.
+    await exec('Discard yarn.lock', `git checkout yarn.lock`);
+
+    // greenkeeper updates each package in the monorepo with the same dependency in different commits: squash and update message
+    await exec('Commit changes and squash.', `git reset --soft HEAD~${COMMITS_TO_SQUASH} && git commit --message="${NEW_COMMIT_MESSAGE}"`);
+
+    await exec(`Update 'origin' remote.`, `git remote remove origin && git remote add origin git@github.com:${GIT_REPO_SLUG}.git`);
+    await exec('Fetch content from origin', 'git fetch origin');
```
Comment (alrra) [inline .travis/greenkeeper-merge.js]: For consistency, use `'origin'`.

File: .travis/greenkeeper-merge.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    await exec('Set remote', `git remote set-branches origin ${GIT_GREENKEEPER_BRANCH}`);
+
+    // Update local data.
+    await exec('Fetch remote', `git fetch`)
+
+    await exec('Configure the Git user name and email.',
+                `git config --global user.name "${GIT_USER_NAME}"
+                 git config --global user.email "${GIT_USER_EMAIL}"`);
+    // `install` step in travis does a yarn. If versions are different, yarn.lock is different so discard changes.
+    await exec('Discard yarn.lock', `git checkout yarn.lock`);
```
Comment (alrra) [inline .travis/greenkeeper-merge.js]: For consistency use `Discard 'yarn.lock'.`.

File: .travis/greenkeeper-merge.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    await exec('Commit changes and squash.', `git reset --soft HEAD~${COMMITS_TO_SQUASH} && git commit --message="${NEW_COMMIT_MESSAGE}"`);
+
+    await exec(`Update 'origin' remote.`, `git remote remove origin && git remote add origin git@github.com:${GIT_REPO_SLUG}.git`);
+    await exec('Fetch content from origin', 'git fetch origin');
+    await exec(`Switch to the 'master' branch.`, 'git checkout master');
+
+    const changes = await exec('List changed files', 'git status');
+    console.log(changes);
+
+    await exec(`Rebase '${GIT_GREENKEEPER_BRANCH}' branch into 'master'`, `git rebase ${GIT_GREENKEEPER_BRANCH}`);
```
Comment (alrra) [inline .travis/greenkeeper-merge.js]: For consistency, add `.` at the end. Same applies to other places.

File: .travis/greenkeeper-merge.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    const GIT_USER_NAME = await getLastCommitAuthorName();
+
+    /*
+     * Ensure that things work as expected for shallow clones.
+     * (This is usually the case with CIs such as Travis)
+     */
+    await exec('Set remote', `git remote set-branches origin ${GIT_GREENKEEPER_BRANCH}`);
+
+    // Update local data.
+    await exec('Fetch remote', `git fetch`)
```
Comment (alrra) [inline .travis/greenkeeper-merge.js]: Missed `;` at the end.

File: .travis/greenkeeper-merge.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+    // greenkeeper updates each package in the monorepo with the same dependency in different commits: squash and update message
+    await exec('Commit changes and squash.', `git reset --soft HEAD~${COMMITS_TO_SQUASH} && git commit --message="${NEW_COMMIT_MESSAGE}"`);
+
+    await exec(`Update 'origin' remote.`, `git remote remove origin && git remote add origin git@github.com:${GIT_REPO_SLUG}.git`);
+    await exec('Fetch content from origin', 'git fetch origin');
+    await exec(`Switch to the 'master' branch.`, 'git checkout master');
+
+    const changes = await exec('List changed files', 'git status');
+    console.log(changes);
```
Comment (alrra) [inline .travis/greenkeeper-merge.js]: Add empty line before this.

**Comment by alrra** [comment]
> although maybe first we want to update the dependencies

Everything that could be update [was updated earlier today](https://github.com/webhintio/hint/commit/db39c3df6736308afd6118f82a330219d7d95f22).

**Pull request closed without merging**